### PR TITLE
refactor!: use `ContractSource` type requirement

### PR DESCRIPTION
**Pull request by antazoey**
### What I did

a `ContractSource` is required anyway, so this changes the type to match. So it is better because it would exit early if for some reason missing some source info and also helps with the project-refactor.

basically, you can't trace a source without a source...

### Checklist

- [ ] All changes are completed
- [ ] New test cases have been added
- [ ] Documentation has been updated

### CODE REVIEW

**Comment by antazoey** [comment]
> ### What I did
> a `ContractSource` is required anyway, so this changes the type to match. So it is better because it would exit early if for some reason missing some source info and also helps with the project-refactor.

basically, you can't trace a source without a source... 
> 
> ### How I did it
> ### How to verify it
> ### Checklist
> * [ ]  All changes are completed
> * [ ]  New test cases have been added
> * [ ]  Documentation has been updated

**Comment by fubuloubu** [comment]
Is this based off sone other large set of changes (or wrong base branch)?

**Comment by antazoey** [comment]
> Is this based off sone other large set of changes (or wrong base branch)?

dang rebasing messed it up

File: src/ape_node/query.py [hunk 1]
```diff
@@ -1,5 +1,6 @@
+from collections.abc import Iterator
```
Comment (fubuloubu) [inline src/ape_node/query.py:1]: is this not in typing?

File: src/ape_ethereum/transactions.py [hunk 2]
```diff
@@ -243,13 +243,14 @@ def method_called(self) -> Optional[MethodABI]:
     @cached_property
     def source_traceback(self) -> SourceTraceback:
```
Comment (fubuloubu) [inline src/ape_ethereum/transactions.py:244]: should this instead be optional, and return None if it can't?

**Review by fubuloubu — APPROVED** [review]

File: src/ape_ethereum/transactions.py (hunk 2 above)
Comment (antazoey) [reply src/ape_ethereum/transactions.py:244]: I think it always can, for the most part. If you don't have tracing, you still get some top-level data.
However, it does fail now and then for some random node-related reasons on providers _with_ tracing support, hence the handling

**Pull request merged**